src/core_plugins/kibana/public/management/sections/indices/create_index_pattern/create_index_pattern.js

```diff
@@ -69,79 +55,81 @@ uiModules.get('apps/management')
+
+        if (dateFields.length === 0) {
+          return {
+            options: [
+              {
+                display: $translate.instant('KIBANA-INDICES_DONT_CONTAIN_TIME_FIELDS')
+              }
+            ]
+          };
+        }
+
+        return {
+          options: [
+            {
+              display: $translate.instant('KIBANA-NO_DATE_FIELD_DESIRED')
+            },
+            ...dateFields.map(field => ({
+              display: field.name,
+              fieldName: field.name
+            })),
+          ]
+        };
+      },
+      err => {
         if (err instanceof IndexPatternMissingIndices) {
-          fetchFieldsError = $translate.instant('KIBANA-INDICES_MATCH_PATTERN');
-          return [];
+          return {
+            error: $translate.instant('KIBANA-INDICES_MATCH_PATTERN')
+          };
         }
 
         throw err;
-      });
-    })
-    .then(fields => {
-      if (fields.length > 0) {
-        fetchFieldsError = null;
-        dateFields = fields.filter(field => field.type === 'date');
       }
-
-      return {
-        fetchFieldsError,
-        dateFields,
-      };
-    }, notify.fatal)
+    )
     .finally(() => {
       loadingCount -= 1;
     });
   };
 
-  const updateFieldList = results => {
-    this.fetchFieldsError = results.fetchFieldsError;
-    if (this.fetchFieldsError) {
-      return;
-    }
+  const findTimeFieldOption = match => {
+    if (!match) return;
 
-    this.dateFields = results.dateFields || [];
-    this.indexHasDateFields = this.dateFields.length > 0;
-    const moreThanOneDateField = this.dateFields.length > 1;
-    if (this.indexHasDateFields) {
-      this.dateFields.unshift(TIME_FILTER_FIELD_OPTIONS.NO_DATE_FIELD_DESIRED);
-    } else {
-      this.dateFields.unshift(TIME_FILTER_FIELD_OPTIONS.NO_DATE_FIELDS_IN_INDICES);
-    }
-
-    if (!moreThanOneDateField) {
-      // At this point the `dateFields` array contains the date fields and the "no selection"
-      // option. When we have less than two date fields we choose the last option, which will
-      // be the "no date fields available" option if there are zero date fields, or the only
-      // date field if there is one.
-      this.formValues.timeField = this.dateFields[this.dateFields.length - 1];
-    }
+    return this.timeFieldOptions.find(option => (
+      // comparison is not done with _.isEqual() because options get a unique
+      // `$$hashKey` tag attached to them by ng-repeat
+      option.fieldName === match.fieldName &&
+        option.display === match.display
+    ));
   };
 
-  const updateFieldListAndSetTimeField = (results, timeFieldName) => {
-    updateFieldList(results);
+  const pickDefaultTimeFieldOption = () => {
+    const noOptions = this.timeFieldOptions.length === 0;
+    const fieldOptions = this.timeFieldOptions.filter(option => !!option.fieldName);
+    const infoOptions = this.timeFieldOptions.filter(option => !option.fieldName);
```

Comment:
Well, they're all selectable, they are really just "non-field" options. I could use that as the description rather than "info" if you think that's better.